src/main/java/hudson/tasks/junit/TestResultAction.java

```diff
@@ -139,14 +145,19 @@ private XmlFile getDataFile() {
         return new XmlFile(XSTREAM, new File(run.getRootDir(), "junitResult.xml"));
     }
 
+    private String getDataFilePath() {
+        return Paths.get(run.getRootDir().getAbsolutePath(), "junitResult.xml").toString();
+    }
+
     @Override
     public synchronized TestResult getResult() {
+        long started = System.currentTimeMillis();
```

Comment:
Here and elsewhere, prefer `System.nanoTime` for measuring elapsed time:

https://www.baeldung.com/java-system-currenttimemillis-vs-system-nanotime

Reply:
Wouldn't that change the behavior to be platform specific?

Reply:
`currentTimeMillis` is wall clock.  if someone or something updates the computers clock whilst this code is running the result will not be the amount of time this method took.  (e.g. NTPd updating the clock by adding 10seconds,  a crude NTP could also set the clock backwards, or a human may just set the date because they noticed it was wrong!).  Thus using this to calculate the "time it took" for something to occur can be (wildly) inaccurate.

`nanoTime` is not related to wall clocks, it is a CPU internal clock.  The only platform specific come here is the resolution of the output.  (but in any case it would be no worse than `currentTimeMillis` 

See the javadoc for this to show how to use it correctly for measuring elapsed time.

Reply:
> Wouldn't that change the behavior to be platform specific?

As explained in https://www.baeldung.com/java-system-currenttimemillis-vs-system-nanotime, the precision would be platform-specific, and the behavior would be correct. With your current implementation the behavior is incorrect.

Reply:
I completely forgot about the wall clock, thanks for reminding me, fixing...

Reply:
Done.